src/Chainweb/Pact/PactService.hs

```diff
@@ -637,10 +643,18 @@ playOneBlock
 playOneBlock currHeader plData pdbenv = do
     miner <- decodeStrictOrThrow' (_minerData $ _payloadDataMiner plData)
     trans <- liftIO $ transactionsFromPayload plData
-    !results <- go miner trans
+
+    -- prop_tx_ttl_validateblock_validate
+    let validTrans = if isGenesisBlock then trans
+          else V.filter k trans
```

Comment:
You can't filter transactions out of the block here. The block payload contains those transactions -- if any transaction in the block fails to validate then block validation should fail for the whole block. This is the reason why we've tried to be so thorough with the validations at newBlock time

Reply:
Also -- transactions at the genesis block have to be valid according to the rules, they aren't special. If they are ill-formed then we need to create well-formed genesis txs, not skip validation here